Applications/Spire/Include/Spire/Charting/Charting.hpp

```diff
@@ -15,11 +16,109 @@ namespace Spire {
   class DropDownColorPicker;
   class StyleDropDownMenuItem;
   class StyleDropDownMenuList;
+  struct TrendLine;
   class TrendLineEditor;
   class TrendLineModel;
   class TrendLineStyleDropDownMenu;
   using Candlestick =
     Nexus::TechnicalAnalysis::Candlestick<ChartValue, ChartValue>;
+
+  //! Returns the slope of the line reprsented by the given points.
+  /*!
+    \param x1 The x-coordinate of the first point.
+    \param y1 The y-cooridnate of the first point.
+    \param x2 The x-coordinate of the second point.
+    \param y2 The y-coordinate of the second point.
+  */
+  template <typename T>
+  T slope(T x1, T y1, T x2, T y2) {
+    if(x1 - x2 == 0) {
```

Comment:
There's no need to check for this.

Reply:
I get division by zero if I don't, should I just throw an exception?

Reply:
Division by zero is exactly what you want right?

Reply:
Won't it just crash? If it's going to have undefined behavior don't I want to define what happens for that case?

Reply:
It won't crash.

Reply:
Okay, I see, it crashed when I was using ints, just saw that x/0 is technically defined when x =/= 0.